### PR TITLE
feat(Table): allow setting any column as row header

### PR DESCRIPTION
**Pull request by marcoskolodny**
Issue: [Link](https://jira.tid.es/browse/WEB-1948)
Specs: [Link](https://www.figma.com/design/WtF16bfvE9kKS4yx3VtLuE/%F0%9F%94%B8-Table-Specs?m=auto&t=mcZHVNAwa2CuBrCz-1)

### CODE REVIEW

File: src/table.tsx
Comment (marcoskolodny) [inline src/table.tsx:1]: Better review [with hidden whitespaces](https://github.com/Telefonica/mistica-web/pull/1175/files?diff=split&w=1)

**Review by Marcosld — APPROVED** [review]

**Review by jpastor-tuenti — APPROVED** [review]

File: src/table.tsx [hunk 1]
```diff
@@ -115,6 +116,7 @@ export const Table = React.forwardRef(
             columnWidth,
             hideHeaders,
             scrollOverResponsiveLayout,
+            rowHeaderIndex = 0,
```
Comment (atabel) [inline src/table.tsx:119]: Does this mean we'll always have a `<th scope="row">`? I think we should support the case of not having one, and I'd say it should be the default case

Comment (marcoskolodny) [reply src/table.tsx]: Discussed offline. We decided to remove default header index and don't show a title in collapsed-rows mode if row's header index has not been specified

**Review by atabel — APPROVED** [review]

**Review by pladaria — APPROVED** [review]

**Review by aweell — APPROVED** [review]

**Pull request merged**